packages/x-charts/src/LineChart/useAreaPlotData.ts

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import { area as d3Area } from '@mui/x-charts-vendor/d3-shape';
+import { useChartGradientIdBuilder } from '../hooks/useChartGradientId';
+import { isBandScale } from '../internals/isBandScale';
+import { ComputedAxisConfig } from '../internals/plugins/featurePlugins/useChartCartesianAxis';
+import { getCurveFactory } from '../internals/getCurve';
+import { ChartsXAxisProps, ChartsYAxisProps } from '../models';
+import { getValueToPositionMapper, useLineSeriesContext, useXAxes, useYAxes } from '../hooks';
+import { DEFAULT_X_AXIS_KEY } from '../constants';
+import { SeriesId } from '../models/seriesType/common';
+
+interface AreaPlotDataPoint {
+  d: string;
+  seriesId: SeriesId;
+  color: string;
+  area?: boolean;
+  gradientId?: string;
+}
+
+export function useAreaPlotData(
+  xAxes: ComputedAxisConfig<ChartsXAxisProps>,
```

Comment:
What about adding the `drawingArea`  as a first arg for coherence with the bar plot.

Could be useful if at some point we need to add algorithm to simplify line shape

Reply:
I could add it, but it wouldn't be used for anything. Would it make sense to add it later when we need it?